File: trip-front/src/api/history.ts
import { get, del } from './request'

export interface TripListItem {
  id: number
  userId: number | null
  fromCity: string | null
  city: string
  days: number
  budget: number
  content: unknown
  status: string
  parentTripId: number | null
  createdAt: string
}

export interface TripDetail extends TripListItem {}

export async function listTrips(page = 1, pageSize = 20) {
  return get<{ items: TripListItem[]; total: number; page: number; pageSize: number }>(
    'history/trips',
    { page, pageSize },
  )
}

export async function getTrip(id: number) {
  return get<TripDetail>(`history/trips/${id}`)
}

export async function deleteTrip(id: number) {
  return del<{ message: string }>(`history/trips/${id}`)
}

File: trip-server-py/src/schemas/history.py
"""History schemas (request/response)"""

from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime
from typing import Optional, List, Dict, Any


class TripResponse(BaseModel):
    """行程响应"""
    model_config = ConfigDict(
        serialize_by_alias=True,
        from_attributes=True
    )
    
    id: int
    user_id: int = Field(alias="userId")
    from_city: Optional[str] = Field(default=None, alias="fromCity")
    city: str
    days: int
    budget: int
    content: Dict[str, Any]
    status: str = "completed"
    parent_trip_id: Optional[int] = Field(default=None, alias="parentTripId")
    created_at: datetime = Field(alias="createdAt")
    updated_at: Optional[datetime] = Field(default=None, alias="updatedAt")


class TripListResponse(BaseModel):
    """行程历史列表响应"""
    model_config = ConfigDict(serialize_by_alias=True)
    
    items: List[TripResponse]
    total: int
    page: int
    page_size: int = Field(alias="pageSize")
